### ISSUE TITLE
[bug]  Issue with TypeScript backend template in version 2.0

### ISSUE THREAD

**Issue by mchao123** [tauri-apps/tauri#7913]
### Describe the bug

I encountered an issue while using the 2.0 beta version of our project. When selecting TypeScript as the backend language, the template does not seem to be correctly created, and the backend code remains in Rust. Additionally, when running the demo, I receive errors in the console.

Could you please take a look and help resolve this issue? Let me know if you need any further information or code snippets. Thank you!
<img width="456" alt="image" src="https://github.com/tauri-apps/tauri/assets/24653378/1f17fe84-7d53-4d68-99d4-d337f105e3fe">

### Platform and versions

```text
[✔] Environment
    - OS: Windows 10.0.22621 X64
    ✔ WebView2: 117.0.2045.43
    ✔ MSVC:
        - Visual Studio Community 2022
        - Visual Studio ���ɹ��� 2022
    ✔ rustc: 1.71.1 (eb26296b5 2023-08-03)
    ✔ Cargo: 1.71.1 (7f1d04c00 2023-07-29)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.2.0
    - pnpm: 8.6.0
    - yarn: 1.22.19
    - npm: 9.6.6
```

**Comment by FabianLars** [comment]
Yeah, it works after upgrading `@tauri-apps/api` - gonna move this issue to the CTA repo, thanks again.
